streams-contrib/streams-provider-youtube/src/main/java/com/youtube/serializer/YoutubeChannelDeserializer.java

```diff
@@ -0,0 +1,131 @@
+package com.youtube.serializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelContentDetails;
+import com.google.api.services.youtube.model.ChannelLocalization;
+import com.google.api.services.youtube.model.ChannelSnippet;
+import com.google.api.services.youtube.model.ChannelStatistics;
+import com.google.api.services.youtube.model.ChannelTopicDetails;
+import com.google.api.services.youtube.model.Thumbnail;
+import com.google.api.services.youtube.model.ThumbnailDetails;
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ *
+ */
+public class YoutubeChannelDeserializer extends JsonDeserializer<Channel> {
+
+
+    @Override
+    public Channel deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+        JsonNode node = jp.getCodec().readTree(jp);
+        try {
+            Channel channel = new Channel();
+            if(node.findPath("etag") != null)
+                channel.setEtag(node.get("etag").asText());
+            if(node.findPath("kind") != null)
+                channel.setKind(node.get("kind").asText());
+            channel.setId(node.get("id").asText());
+            channel.setTopicDetails(setTopicDetails(node.findValue("topicDetails")));
+            channel.setStatistics(setChannelStatistics(node.findValue("statistics")));
+            channel.setContentDetails(setContentDetails(node.findValue("contentDetails")));
+            channel.setSnippet(setChannelSnippet(node.findValue("snippet")));
+            return channel;
+        } catch (Throwable t) {
+            throw new IOException(t);
+        }
```

Comment:
In the event that there is a deserialization error, could we log the error message and return a null object? I'm not sure we want to derail the entire operation if one datum has trouble getting deserialized.